tests/driver_lis2dh12/main.c

```diff
@@ -11,170 +11,699 @@
-        };
-        lis2dh12_set_int(&dev, &params_int2, LIS2DH12_INT2);
+    if (gpio_init_int(pin1, GPIO_IN, GPIO_RISING, lis2dh12_int_cb, &line1)) {
+        DEBUG("[lis_init]: INT1 failed\n");
+    }
+    else {
+        DEBUG("[lis_init]: INT1 done\n");
     }
-#endif
 
-    while (1) {
+    if (gpio_init_int(pin2, GPIO_IN, GPIO_RISING, lis2dh12_int_cb, &line2)) {
+        DEBUG("[lis_init]: INT2 failed\n");
+    }
+    else {
+        DEBUG("[lis_init]: INT2 done\n");
+    }
+#endif /* MODULE_LIS2DH12_INT */
+
+    /* enable FIFO */
+    lis2dh12_set_fifo(&dev, &fifo_cfg);
+
+    /* enable click detection */
+    lis2dh12_set_click(&dev, &click_cfg);
+
+    /* set default shock value */
+    shock_thold = THOLD_SHOCK_MILLIG_DEFAULT;
+
+    /* read registers to reset device */
+    lis2dh12_read_click_src(&dev, &click_src_reg);
+    lis2dh12_read_reference(&dev, &reference_value);
+#ifdef MODULE_LIS2DH12_INT
+    lis2dh12_read_int_src(&dev, &int1_src, 1);
+#endif /* MODULE_LIS2DH12_INT */
+}
 
 #ifdef MODULE_LIS2DH12_INT
-        if (xtimer_mutex_lock_timeout(&isr_mtx, DELAY) == 0) {
-            flags = isr_flags;
-            isr_flags = 0;
+void* lis2dh12_test_process(void* arg) {
+    (void) arg;
+    while (1) {
+        /* start processing */
+        DEBUG("[Process]: start process\n");
+
+        /* read FIFO_src before getting data */
+        LIS2DH12_FIFO_SRC_REG_t fifo_src;
+        lis2dh12_read_fifo_src(&dev, &fifo_src);
+        DEBUG("[Process]: FIFO SRC 0x%x\n", fifo_src.reg);
+        DEBUG("[Process]: WTM %x, OVRN %d, EMPTY %d, FSS %d\n", fifo_src.bit.WTM,
+                                fifo_src.bit.OVRN_FIFO, fifo_src.bit.EMPTY, fifo_src.bit.FSS);
+
+        /* get fifo data */
+        uint8_t number_read = lis2dh12_read_fifo_data(&dev, data_fifo, NUM_FIFO_VALUES);
+
+        /* read FIFO_src after getting data */
+        lis2dh12_read_fifo_src(&dev, &fifo_src);
+        DEBUG("[Process]: FIFO SRC 0x%x\n", fifo_src.reg);
+        DEBUG("[Process]: WTM %x, OVRN %d, EMPTY %d, FSS %d\n", fifo_src.bit.WTM,
+                                fifo_src.bit.OVRN_FIFO, fifo_src.bit.EMPTY, fifo_src.bit.FSS);
+
+        /* display FIFO data */
+        if (ENABLE_DEBUG) {
+            for (int i = 0; i < number_read; i++){
+                printf("[Process]: X[%2d]  %d\n", i, data_fifo[i].X_AXIS);
+                printf("[Process]: Y[%2d]  %d\n", i, data_fifo[i].Y_AXIS);
+                printf("[Process]: Z[%2d]  %d\n", i, data_fifo[i].Z_AXIS);
+            }
+        }
+
+        /* After the Interrupt the FIFO needs to be enabled again. */
+        lis2dh12_restart_fifo(&dev);
+
+        /* check if shock occurred*/
+        uint16_t max_data_X = 0;
+        uint16_t max_data_Y = 0;
+        uint16_t max_data_Z = 0;
+
+        bool X_shock_pos = false;
+        bool Y_shock_pos = false;
+        bool Z_shock_pos = false;
+
+        for (uint8_t entry = NUM_FIFO_VALUES - NUM_DATA_SHOCK_DETECT; entry < NUM_FIFO_VALUES;
```

Comment:
shouldn't this be

```suggestion
        for (uint8_t entry = number_read - NUM_DATA_SHOCK_DETECT; entry < number_read;
```
?

Reply:
yes. should I do a new PR for all this?

Reply:
I'm already at it, since I was looking at how to use this for the app integration (yea should have done so earlier…)


Btw, does the zero point calibration (that was what `old_data_lis` was supposed to be for I guess) really need to be done externally?

Reply:
I used `old_data_lis` in the shock detection, to subtract the privious value. I think it is needed to have more precise values and better detect the position/rotation. The zero point calib would be done onces and if position is changed, than you get unknown data.